securedrop/tests/test_journalist_api.py

```diff
@@ -0,0 +1,596 @@
+# -*- coding: utf-8 -*-
+import hashlib
+import json
+import os
+
+from pyotp import TOTP
+
+from flask import current_app, url_for
+from itsdangerous import TimedJSONWebSignatureSerializer
+
+from db import db
+from models import Journalist, Reply, Source, SourceStar, Submission
+
+os.environ['SECUREDROP_ENV'] = 'test'  # noqa
+from utils.api_helper import get_api_headers
+
+
+def test_unauthenticated_user_gets_all_endpoints(journalist_app):
+    with journalist_app.test_client() as app:
+        response = app.get(url_for('api.get_endpoints'))
+
+        observed_endpoints = json.loads(response.data)
+
+        for expected_endpoint in ['current_user_url', 'sources_url',
```

Comment:
Nit: This allows for unknown endpoints to be present. Maybe we check that a known list `==` the keys?